### PR TITLE
[NFC] Sema: get rid of redundant DeclContext parameter to checkConfor…

### PR DESCRIPTION
**Pull request by AnthonyLatsis**
…mancesInContext

Follow-up to #31645

### CODE REVIEW

File: lib/Sema/TypeCheckProtocol.cpp [hunk 1]
```diff
@@ -5034,24 +5034,21 @@ LookupAllConformancesInContextRequest::evaluate(
...
+void TypeChecker::checkConformancesInContext(IterableDeclContext *idc) {
+  auto *const dc = idc->getAsGenericContext();
+
   // For anything imported from Clang, lazily check conformances.
   if (isa<ClangModuleUnit>(dc->getModuleScopeContext()))
     return;
 
   // Determine the access level of this conformance.
-  Decl *currentDecl = nullptr;
   AccessLevel defaultAccess;
```
Comment (CodaFi) [inline lib/Sema/TypeCheckProtocol.cpp:5045]: Let's just get rid of this braced block and compute `defaultAccess` below the conditional.

Comment (AnthonyLatsis) [reply lib/Sema/TypeCheckProtocol.cpp:5045]: Do you mean
```cpp
AccessLevel defaultAccess;
if (auto nominal = dc->getSelfNominalDecl()) {
  defaultAccess = nominal->getFormalAccess();
} else {
  return;
}
```
?

Comment (CodaFi) [reply lib/Sema/TypeCheckProtocol.cpp:5045]:
```
const auto *nominal = dc->getSelfNominalTypeDecl();
if (!nominal)
  return;
auto defaultAccess = nominal->getFormalAccess();
```

Comment (AnthonyLatsis) [reply lib/Sema/TypeCheckProtocol.cpp]: Oh, the conditional *return*

**Review by CodaFi — APPROVED** [review]

**Comment by CodaFi** [comment]
@swift-ci smoke test

**Review by slavapestov — APPROVED** [review]

**Pull request merged**